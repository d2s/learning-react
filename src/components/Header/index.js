import React from 'react'
import HeaderLink from '../HeaderLink'

export default (props) => (
  <header className='tc pv4 pv5-ns'>
    <nav className="pa3 pa4-ns">
      <HeaderLink name="Site Name" />
      <HeaderLink name="Home" />
      <HeaderLink name="About" />
      <HeaderLink name="Store" />
      <HeaderLink name="Contact" />
    </nav>
  </header>
)
