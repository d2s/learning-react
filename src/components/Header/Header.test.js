import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'

import Header from './index';

describe('With ReactDOM', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header />, div);
  });
})

describe('With Enzyme', () => {
  it('shallow renders without crashing', () => {
    shallow(<Header />);
  });
})

describe('With Snapshot Testing', () => {
  it('Header renders the same way', () => {
    const component = renderer.create(<Header />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
