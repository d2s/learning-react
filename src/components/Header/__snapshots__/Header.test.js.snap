exports[`With Snapshot Testing Header renders the same way 1`] = `
<header
  className="tc pv4 pv5-ns">
  <nav
    className="pa3 pa4-ns">
    <a
      className="link dim gray f6 f5-ns dib mr3"
      href="#"
      title="Site Name">
      Site Name
    </a>
    <a
      className="link dim gray f6 f5-ns dib mr3"
      href="#"
      title="Home">
      Home
    </a>
    <a
      className="link dim gray f6 f5-ns dib mr3"
      href="#"
      title="About">
      About
    </a>
    <a
      className="link dim gray f6 f5-ns dib mr3"
      href="#"
      title="Store">
      Store
    </a>
    <a
      className="link dim gray f6 f5-ns dib mr3"
      href="#"
      title="Contact">
      Contact
    </a>
  </nav>
</header>
`;
